### PR TITLE
Add counter value when triggering ROTP#provisioning_uri

### PR DESCRIPTION
**Pull request by wenderjean**
Related to: https://github.com/heapsource/active_model_otp/issues/74

## Changes

- Add `self.otp_counter` when calling `ROTP#provisioning_uri` in order to have it on the final URI

## Concerns

- Even though it seems somethings that never caused issues, reading through the [RFC-4226](https://tools.ietf.org/html/rfc4226#section-7.4) I noticed some requirements on keeping the counter synchronized between server/client, please, let me know your thoughts regarding that.

Note.: **_Taking a look at the source it doesn't seem we're caring about the `houndci-bot` for line length metrics. Did we enable `houndci-bot` recently?_**

### CODE REVIEW

File: active_model_otp.gemspec [hunk 1]
```diff
@@ -24,6 +24,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency "rotp", "~> 5.0.0"
 
   spec.add_development_dependency "activerecord"
+  spec.add_development_dependency "activemodel-serializers-xml"
```
Comment (wenderjean) [inline active_model_otp.gemspec:27]: The motivation behind this change is the error I've been facing during the development process.
```
.../active_model_otp/test/models/user.rb:4:in `<class:User>': uninitialized constant ActiveModel::Serializers::Xml (NameError)
```

**Comment by pedrofurtado** [comment]
Hi @wenderjean ! 👋 

Thanks for your contribution (and delayed review of my part) 🤝 🍻 

That's a great fix! In fact, when we generate the provisioning uri with otp counter previously used (and then with a value greater than zero), the authenticator app and server/backend be in sync.

Please, could you fix the CI warnings? After that, @robertomiranda , I think we are ready to merge it ✅ 🏁

**Comment by wenderjean** [comment]
I'll take a look on tests asap guys, I'm using a different PC right now for that reason I'm getting those issues.

**Comment by pedrofurtado** [comment]
Thanks a lot @wenderjean 🍻

File: test/one_time_password_test.rb [hunk 2]
```diff
@@ -100,22 +102,51 @@ def test_otp_code_without_specific_length
...
   end
 
   def test_provisioning_uri_with_options
-    assert_match %r{^otpauth://totp/Example\:roberto%40heapsource\.com\?secret=\w{32}&issuer=Example$}, @user.provisioning_uri(nil, issuer: "Example")
-    assert_match %r{^otpauth://totp/Example\:roberto%40heapsource\.com\?secret=\w{32}&issuer=Example$}, @visitor.provisioning_uri(nil, issuer: "Example")
-    assert_match %r{^otpauth://totp/Example\:roberto\?secret=\w{32}&issuer=Example$}, @user.provisioning_uri("roberto", issuer: "Example")
-    assert_match %r{^otpauth://totp/Example\:roberto\?secret=\w{32}&issuer=Example$}, @visitor.provisioning_uri("roberto", issuer: "Example")
+    account = %r{
+      ^otpauth://totp/Example\:roberto\?secret=\w{32}&issuer=Example$
+    }x
```
Comment (wenderjean) [inline test/one_time_password_test.rb:125]: The trick for multi-line regex content: `}x`

**Comment by wenderjean** [comment]
All good @pedrofurtado :)

**Comment by pedrofurtado** [comment]
Thank for your effort on it, @wenderjean ! 🤝 

Our team is planning (especially me and @robertomiranda) to release this and some other improvements/bugfixes in a new release soon 🎉

**Pull request merged**